### PR TITLE
Fix/1856

### LINKED ISSUE #1856: CustomPomParser produces unparsable ivy descriptor when POM has a parent
#1191 is closed, so I'm opening this because it's still an issue.

If a child POM has a parent section, the transformed child will have an e:sbtTransformHash tag without the corresponding namespace declaration.

Here's why:
- The parent POM is processed and addExtra is used.  This adds e:sbtTransformHash to its extraInfo and adds the extra namespace
- Ivy's PomModuleDescriptorParser adds the parent's extra info to the child before finishing parseDescriptor
- CustomPomParser begins transforming the child module, but it already has the sbtTransformHash tag, so it skips addExtra

Because addExtra is never run on the child module, the child is left with the extraInfo sbtTransformHash tag but never gets the addExtraNamespace.

### PR DESCRIPTION
**Pull request by jsuereth**
Fixes #1856 

REview by @eed3si9n

### CODE REVIEW

**Comment by eed3si9n** [comment]
This hopefully fixes news but do you need to do anything about the existing/cached poms that broke?

**Comment by jsuereth** [comment]
@eed3si9n Comments accounted for, and rebased.

File: ivy/src/main/scala/sbt/CustomPomParser.scala [hunk 1]
```diff
@@ -95,10 +108,10 @@ object CustomPomParser {
       val mergeDuplicates = IvySbt.hasDuplicateDependencies(md.getDependencies)
 
       val unqualify = toUnqualify(filtered)
-      if (unqualify.isEmpty && extraDepAttributes.isEmpty && !convertArtifacts && !mergeDuplicates)
```
Comment (dwijnand) [inline ivy/src/main/scala/sbt/CustomPomParser.scala:98]: Should we kill `mergeDuplicates`, `convertArtifacts` and the `artifactExtIncorrect` method as a part of this PR?

File: ivy/src/main/scala/sbt/CustomPomParser.scala [hunk 2]
```diff
@@ -65,11 +73,16 @@ object CustomPomParser {
...
+      val h = MyHash
       // sbt 0.13.1 used "sbtTransformHash" instead of "e:sbtTransformHash" until #1192 so read both
       Option(extraInfo).isDefined &&
         ((Option(extraInfo get TransformedHashKey) orElse Option(extraInfo get oldTransformedHashKey)) match {
-          case Some(TransformHash) => true
-          case _                   => false
+          case x @ Some(MyHash) =>
+            true
+          case Some(other) =>
+            false
```
Comment (dwijnand) [inline ivy/src/main/scala/sbt/CustomPomParser.scala]: redundant?

File: ivy/src/main/scala/sbt/CustomPomParser.scala [hunk 3]
```diff
@@ -65,11 +73,16 @@ object CustomPomParser {
     {
       val oldTransformedHashKey = "sbtTransformHash"
       val extraInfo = md.getExtraInfo
+      val MyHash = MakeTransformHash(md)
+      val h = MyHash
```
Comment (dwijnand) [inline ivy/src/main/scala/sbt/CustomPomParser.scala]: redundant?

**Comment by dwijnand** [comment]
I remember trying to figure out how to fix this. Nice work.

File: ivy/src/main/scala/sbt/CustomPomParser.scala (hunk 3 above)
Comment (jsuereth) [inline ivy/src/main/scala/sbt/CustomPomParser.scala]: ah, wow.  Yeah, leaving my debug mess hanging around.

File: ivy/src/main/scala/sbt/CustomPomParser.scala (hunk 1 above)
Comment (jsuereth) [inline ivy/src/main/scala/sbt/CustomPomParser.scala:98]: not in this PR.  mergeDuplicates has some work it still does which is useful, we just need to figure otu either an alternative mechanism _or_ make that mechanism work.

**Comment by eed3si9n** [comment]
LGTM. notes plz.

**Comment by jsuereth** [comment]
Notes there.  You can wait for the next build, or merge directly.

**Comment by eed3si9n** [comment]
Merged!

**Pull request merged**